File: workspace-raspi/http_pic_t.v1.3.2.1/src/serverConfig.c
/*
 * serverConfig.c
 *	v1.3.2
 *  update on: 2015/11/8
 *      Author: nishi
 */

#define SERVERCONFIG_DEF

#include <pthread.h>
#include <fcntl.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>


#include "tableDef.h"
#include "serverConfig.h"
#include "com_lib.h"
#include "checkAllowNetworks.h"

Defs defs;
pthread_mutex_t mutex_mem;

void init_serverConfig(Defs *defs){
	defs->version = "1.3.2.1";	// Version
	defs->update = "2015.11.11";	// Update
	defs->HTTP_PORT = 80;	//HTTP Server connection port no
	defs->conf_file = "server.conf";
	defs->d_conf_file = "/etc/http_pid/server.conf";	// conf_file for deamon
	defs->myhost = "www6.net-tosa.cxm";
	defs->myhostip ="192.168.1.150";
	defs->myuser="nishi";	// nishi / pi-nishi
	defs->timeout=120;	// [sec]
	defs->maxConnect=5;
	defs->maxThread_cnt= defs->maxConnect+1;	// Max Tread count = maxConnect+1
	defs->allow_networks="192.168.1.0/255.255.255.0";

	defs->doc_rootDir ="/home/not-use"; // "/home/nishi/www-raspi/html"

	defs->cgi_rootDir ="/home/not-use"; // "/home/nishi/www-raspi/cgi-bin"
	defs->ccgi_rootDir ="/home/not-use"; // "/home/nishi/www-raspi/ccgi-binc"	// add by nishi 2015.11.8

	defs->class_rootDir ="/home/not-use"; // "/home/nishi/workspace-raspi/nishi-test/bin"
	defs->class_modDir ="/home/not-use"; // "/home/nishi/workspace-raspi/nishi-test/bin"
	defs->tmp_dir ="/tmp/ras-pi-upload";	// file up-load working dir
		// Activated value
	defs->error_log="";
	defs->built_in="";		// built-in package name
	defs->class_path="";	// ".:/home/nishi/workspace-raspi/nishi-test2/bin:/usr/java/default/jre/lib/rt.jar"
}

void serverConfig(Defs *defs){
	//printf("serverConfig():#1 start\n");
	printf("--- %s define\n",defs->conf_file);

	FILE *fp = fopen(defs->conf_file,"r");
	size_t r_lng;
	char i_buff[256];
	char wk[128];
	if(fp != NULL){
		while(fgets(i_buff,sizeof(i_buff),fp) != NULL){
			rtrim(i_buff,0x0a);
			rtrim(i_buff,0x0d);
			rtrim(i_buff,' ');
			if(str_startsWith(i_buff,"http_port=")==0){
				strcpy(wk,i_buff+10);
				defs->HTTP_PORT = atoi(wk);
				printf("http_port=%d\n",defs->HTTP_PORT);
			}
			else if(str_startsWith(i_buff,"timeout=")==0){
				strcpy(wk,i_buff+8);
				defs->timeout = atoi(wk);
				printf("timeout=%d\n",defs->timeout);
			}
			else if(str_startsWith(i_buff,"maxconnect=")==0){
				strcpy(wk,i_buff+11);
				defs->maxConnect = atoi(wk);
				//defs->maxThread_cnt= (defs->maxConnect)+1;
				defs->maxThread_cnt= (defs->maxConnect);	// 試験用に+1しません。
				printf("maxconnect=%d\n",defs->maxConnect);
			}
			else if(str_startsWith(i_buff,"doc_rootdir=")==0){
				defs->doc_rootDir=get_strcpy(i_buff+12);
				printf("doc_rootdir=%s\n",defs->doc_rootDir);
			}
			else if(str_startsWith(i_buff,"cgi_rootdir=")==0){
				defs->cgi_rootDir=get_strcpy(i_buff+12);
				printf("cgi_rootdir=%s\n",defs->cgi_rootDir);
			}
			else if(str_startsWith(i_buff,"ccgi_rootdir=")==0){
				defs->ccgi_rootDir=get_strcpy(i_buff+13);
				printf("ccgi_rootdir=%s\n",defs->ccgi_rootDir);
			}
			else if(str_startsWith(i_buff,"allow_networks=")==0){
				defs->allow_networks=get_strcpy(i_buff+15);
				printf("allow_networks=%s\n",defs->allow_networks);
			}
			// class_rootdir= add by nishi 2015.10.27
			else if(str_startsWith(i_buff,"class_rootdir=")==0){
				defs->class_rootDir=get_strcpy(i_buff+14);
				printf("class_rootDir=%s\n",defs->class_rootDir);
			}
			// class_rootdir= add by nishi 2015.11.11
			else if(str_startsWith(i_buff,"class_moddir=")==0){
				defs->class_modDir=get_strcpy(i_buff+13);
				printf("class_modDir=%s\n",defs->class_modDir);
			}
			// class_path= add by nishi 2015.11.15
			else if(str_startsWith(i_buff,"class_path=")==0){
				defs->class_path=get_strcpy(i_buff+11);
				printf("class_path=%s\n",defs->class_path);
			}
		}
		fclose(fp);

		checkAllowNetworks_init(defs->allow_networks);
	}
	else{
		printf("serverConfig():#90 %s open error\n",defs->conf_file);

	}
}

File: workspace-raspi/net/src/http_pi/serverConfig.java
/**
 * Raspberry Pi java http server
 * v1.7 update 2015.1.17
 * serverConfig.java : プログラム実行パラメータの取得。
 * server.conf を読み込みます
 */
package http_pi;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;

/**
 * @author nishi
 *
 */
public class serverConfig {
	/**
	 * 
	 */
	private boolean deamon_f;
	public serverConfig(boolean deamon_f) {
		this.deamon_f=deamon_f;
	}
	public void readConf(){
		File file;
		// TODO 自動生成されたコンストラクター・スタブ
		if(deamon_f==true){
			file = new File(Defs.d_conf_file);
		}
		else{
			file = new File(Defs.conf_file);
		}
	    //ファイルを読み込む処理。
	    BufferedReader rd;
	    String s,rec;
	    try{
	      //ファイルを開く
	      rd=new BufferedReader(new FileReader(file));
	      //1行ずつ読み込む
	      for (rec=rd.readLine();rec!=null;rec=rd.readLine()){
	    	  if(rec=="" || rec.startsWith("#") == true){
	    		  continue;
	    	  }
	    	  if(rec.startsWith("http_port=")==true){
	    		  s=rec.substring("http_port=".length());
	    		  Defs.HTTP_PORT=Integer.parseInt(s);
	    	  }
	    	  else if(rec.startsWith("myhost=")==true){
	    		  Defs.myhost=rec.substring("myhost=".length());
	    	  }
	    	  else if(rec.startsWith("myhostip=")==true){
	    		  Defs.myhostip=rec.substring("myhostip=".length());
	    	  }
	    	  else if(rec.startsWith("maxthread_cnt=")==true){
	    		  s=rec.substring("maxthread_cnt=".length());
	    		  Defs.maxThread_cnt=Integer.parseInt(s);
	    	  }
	    	  else if(rec.startsWith("allow_networks=")==true){
	    		  Defs.allow_networks=rec.substring("allow_networks=".length());
	    	  }
	    	  else if(rec.startsWith("doc_rootdir=")==true){
	    		  Defs.doc_rootDir=rec.substring("doc_rootdir=".length());
	    	  }
	    	  else if(rec.startsWith("cgi_rootdir=")==true){
	    		  Defs.cgi_rootDir=rec.substring("cgi_rootdir=".length());
	    	  }
	    	  else if(rec.startsWith("class_rootdir=")==true){
	    		  Defs.class_rootDir=rec.substring("class_rootdir=".length());
	    	  }
	    	  else if(rec.startsWith("class_moddir=")==true){
	    		  Defs.class_modDir=rec.substring("class_moddir=".length());
	    	  }
	    	  else if(rec.startsWith("tmp_dir=")==true){
	    		  Defs.tmp_dir=rec.substring("tmp_dir=".length());
	    	  }
	    	  else if(rec.startsWith("error_log=")==true){
	    		  Defs.error_log=rec.substring("error_log=".length());
	    	  }
	    	  else if(rec.startsWith("built_in=")==true){
	    		  Defs.built_in=rec.substring("built_in=".length()).trim();
	    		  Defs.built_in_cl = Class.forName(Defs.built_in);
	    	  }
	    	  else if(rec.startsWith("class_interface=")==true){
	    		  s=rec.substring("class_interface=".length());
	    		  Defs.class_interface=Integer.parseInt(s);
	    	  }
	    	  // v1.7
	    	  else if(rec.startsWith("with_php=")==true){
	    		  s=rec.substring("with_php=".length());
	    		  int i = Integer.parseInt(s);
	    		  if(i==0){
		    		  Defs.with_php=false;
	    		  }
	    		  else if(i==1){
		    		  Defs.with_php=true;
	    		  }
	    	  }
	    	  // v1.7
	    	  else if(rec.startsWith("php_cgi=")==true){
	    		  Defs.php_cgi=rec.substring("php_cgi=".length());
	    	  }
	    	  // v1.8
	    	  else if(rec.startsWith("KA=")==true){
	    		  s=rec.substring("KA=".length());
	    		  if(s.equals("off")){
	    			  Defs.KA=0;
	    		  }
	    		  else if(s.equals("on")){
	    			  Defs.KA=1;
	    		  }
	    		  else if(s.equals("mix")){
	    			  Defs.KA=2;
	    		  }
	    	  }
	    	  // v1.8
	    	  else if(rec.startsWith("MKAR=")==true){
	    		  s=rec.substring("MKAR=".length());
	    		  Defs.MKAR=Integer.parseInt(s);
	    	  }
	    	  // v1.8
	    	  else if(rec.startsWith("KAT=")==true){
	    		  s=rec.substring("KAT=".length());
	    		  Defs.KAT=Integer.parseInt(s);
	    	  }
	    	  // v1.8
	    	  else if(rec.startsWith("CIBL=")==true){
	    		  s=rec.substring("CIBL=".length());
	    		  Defs.CIBL=Integer.parseInt(s);
	    	  }
	    	  // v1.8
	    	  else if(rec.startsWith("Expi=")==true){
	    		  s=rec.substring("Expi=".length());
	    		  Defs.Expi=Integer.parseInt(s)*60*1000;
	    	  }
	    	  // v1.8.1
	    	  else if(rec.startsWith("Timeout=")==true){
	    		  s=rec.substring("Timeout=".length());
	    		  Defs.Timeout=Integer.parseInt(s);
	    	  }
	      }
	      rd.close();
	    }
		catch(FileNotFoundException e){
			System.err.println("serverConfig::readConf() :#9 server.conf file nothing!!");
			System.exit(1);
		}
	    catch(IOException e){
			System.err.println("serverConfig::readConf() :#10 server.conf file access error e="+e);
			System.exit(2);
	    } catch (ClassNotFoundException e) {
			// TODO 自動生成された catch ブロック
			//e.printStackTrace();
			System.err.println("serverConfig::readConf() :#11 server.conf Calss.forName error e="+e);
			System.exit(3);
		}
	}
}
